pghoard/rohmu/object_storage/azure.py

```diff
@@ -165,59 +182,62 @@ def _stream_blob(self, key, fileobj, progress_callback):
         allows reading entire blob into memory at once or returning data from random offsets"""
         file_size = None
         start_range = 0
-        chunk_size = self.conn.MAX_CHUNK_GET_SIZE
+        chunk_size = self.conn._config.max_chunk_get_size  # pylint: disable=protected-access
         end_range = chunk_size - 1
         while True:
             try:
                 # pylint: disable=protected-access
-                blob = self.conn._get_blob(self.container_name, key, start_range=start_range, end_range=end_range)
+                blob = self.conn.get_blob_client(self.container_name, key)
+                blob_properties = blob.get_blob_properties()
                 if file_size is None:
-                    file_size = self._parse_length_from_content_range(blob.properties.content_range)
-                fileobj.write(blob.content)
-                start_range += blob.properties.content_length
+                    file_size = blob_properties.size
+                download_stream = blob.download_blob()
+                fileobj.write(download_stream.readall())
```

Comment:
This looks like it will just read the entire file into memory. Nothing here makes use of the loop, start_range or end_range. I presume just changing this to `readinto` would make this work in the sense that it wouldn't just run out of memory when trying to download very large file but it would be extremely nice to have progress reporting working as we have some cases where we're downloading files that are tens of gigabytes or more in size and not having progress printed out could be quite inconvenient.

Reply:
Ah yes,  I had this on my todo list to go back and fix as I was working, but it slipped through the cracks. Fixed now, good catch.